File: turkisheco-frontend/src/app/models/comment.ts
export interface Comment {
  id: number;
  postId: number;
  authorName: string;
  authorEmail?: string | null;
  content: string;
  createdAt: string;
  isPendingModeration?: boolean;
}

export interface CreateCommentRequest {
  authorName: string;
  authorEmail?: string | null;
  content: string;
  website?: string | null;
}

File: backend/Turkisheco.Api/Entities/Comment.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Turkisheco.Api.Entities
{
    public class Comment
    {
        public int Id { get; set; }

        [Required]
        public int PostId { get; set; }
        public Post Post { get; set; } = null!;

        public int? ForumUserId { get; set; }
        public ForumUser? ForumUser { get; set; }

        public string? AuthorName { get; set; }
        public string? AuthorEmail { get; set; }

        [Required]
        [MaxLength(2000)]
        public string Content { get; set; } = string.Empty;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}
